File: nodejs/get_odds.js

// get odds from list of integers

var nums = process.argv.slice(2).map(function(argItem) {
    return parseInt(argItem);
});

var odds = nums.filter(function(num) {
    return num % 2 === 1 || num % 2 === -1;
});

console.log(odds.join(' '));

File: golang/get_odds.go

 // get odds from list of integers

package main

import (
    "os"
    "fmt"
    "strconv"
    "strings"
)


func main() {

    nums := os.Args[1:]
    var odds []string

    for _, element := range nums {
        curr_int, err := strconv.ParseInt(element, 10, 64)
        if err == nil && curr_int % 2 != 0 {
            odds = append(odds, element)
        }
    }

    fmt.Println(strings.Join(odds[:], " "))

}
